var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -172,22 +172,15 @@ def setup_build_environment(self, env):
 
         # ESMF_COMPILER must be set to select which Fortran and
         # C++ compilers are being used to build the ESMF library.
+        print(f"compiler is {self.compiler.name}")
         if self.compiler.name == "gcc":
             env.set("ESMF_COMPILER", "gfortran")
-            gfortran_major_version = int(
-                spack.compiler.get_compiler_version_output(self.compiler.fc, "-dumpversion").split(
-                    "."
-                )[0]
-            )
+            gfortran_major_version = self.compiler.real_version[0]
```

Comment:
Does this work if the C compiler is clang and the Fortran compiler is gfortran? This is the reason why line 177 was introduced in the first place. `clang` may be version `14.0.0`, but gfortran may be `9.2.0`.

Reply:
I'm not sure if I can test that on my system - can you try?

Reply:
That is, we need to get the Fortran compiler version, not the spack (= usually the C) compiler version. Here is a typical compiler setup for macOS, where clang+gfortran is recommended (by the spack developers)
```
[skylab-env] heinzell@JCSDA-L-18146:~/prod/spack-stack-1.3.0 [brew-x86_64]> cat envs/unified-env/site/compilers.yaml
compilers:
- compiler:
    spec: apple-clang@13.1.6
    paths:
      cc: /usr/bin/clang
      cxx: /usr/bin/clang++
      f77: /usr/local/bin/gfortran
      fc: /usr/local/bin/gfortran
    flags: {}
    operating_system: monterey
    target: x86_64
    modules: []
    environment: {}
    extra_rpaths: []
```

Reply:
I think that we found a way that it works for everyone, @AlexanderRichert-NOAA can you test again?